src/python/pants/engine/internals/options_parsing.py

```diff
@@ -60,16 +58,6 @@ def log_level(global_options: GlobalOptions) -> LogLevel:
     return global_options.level
 
 
-@rule
-def extract_process_cleanup_option(keep_sandboxes: KeepSandboxes) -> ProcessCleanupOption:
-    warn_or_error(
-        removal_version="2.15.0.dev1",
-        entity="ProcessCleanupOption",
-        hint="Instead, use `KeepSandboxes`.",
-    )
-    return ProcessCleanupOption(keep_sandboxes == KeepSandboxes.never)
```

Comment:
This copy of the deprecation was meant for plugin authors who might be consuming the `ProcessCleanupOption` type. Not critical though.

Reply:
Aha, that explains the strange spelling. I could not for the life of me understand why you used internal names like `KeepSandboxes` when a user never sees that. Now it makes sense. Are API deprecations a thing generally as we edge closer or were you one off experimenting with this?